### DISCUSSION TITLE
Is it possible to create multiple rows with createMany when the values of one of the columns are in a list?

### DISCUSSION THREAD

**Discussion by talonx** [prisma/prisma#24588] [Q&A]
### Question

The relevant model is
```prisma
model UserFeature {
  id        Int     @id @default(autoincrement())
  featureId Int
  userId    String
  user      User    @relation(fields: [userId], references: [id], onDelete: Cascade)
  feature   Feature @relation(fields: [featureId], references: [id], onDelete: Cascade)
}
```
I want to insert multiple rows for a single userId and a list of `featureId`s. Is there a way to do this with a createMany or do I have to use multiple create calls - one for each featureId in the list?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @talonx 👋 

Thank you for raising this question.

According to the [documentation](https://www.prisma.io/docs/orm/prisma-client/queries/relation-queries#create-multiple-records-and-multiple-related-records), you cannot access relations in a `createMany` query. You would need to use multiple create calls to achieve this. For example:
```js
async function addUserFeatures(prisma, userId, featureIds) {
  const inserts = featureIds.map(featureId => 
    prisma.userFeature.create({
      data: {
        userId: userId,
        featureId: featureId
      }
    })
  );
  await prisma.$transaction(inserts);
}
```
This function maps over the list of `featureIds` and creates a `userFeature` entry for each `featureId` associated with the given `userId`. It then executes all the `create` operations in a transaction to ensure atomicity.

There is also an open [issue](https://github.com/prisma/prisma/issues/5455) that hopes to address this in the future.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
